.github/workflows/ci.yml

```diff
@@ -0,0 +1,239 @@
+    branches:
+      - master
+jobs:
+  build:
+    name: Build image
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+
+      - name: Install click
+        run: |
+          python -m pip install --upgrade pip
+          pip install click pyyaml requests
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # 0 is "all history and branch tags"
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.FM_DOCKERHUB_TOKEN_USER }}
+          password: ${{ secrets.FM_DOCKERHUB_TOKEN }}
+
+      - name: Prepare lite-build image
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG build-build-image \
+          --pull-image \
+          --push-image
+        env:
+          QSERV_DH_USER: ${{ secrets.FM_DOCKERHUB_TOKEN_USER }}
+          QSERV_DH_TOKEN: ${{ secrets.FM_DOCKERHUB_TOKEN }}
+
+      - name: Prepare lite-run-base image
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG build-run-base-image \
+          --pull-image \
+          --push-image
+        env:
+          QSERV_DH_USER: ${{ secrets.FM_DOCKERHUB_TOKEN_USER }}
+          QSERV_DH_TOKEN: ${{ secrets.FM_DOCKERHUB_TOKEN }}
+
+      - name: Prepare user build image
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG build-user-build-image \
+          --group docker_outer
+
+      - name: Prepare maraidb image
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG build-mariadb-image \
+          --pull-image \
+          --push-image
+        env:
+          QSERV_DH_USER: ${{ secrets.FM_DOCKERHUB_TOKEN_USER }}
+          QSERV_DH_TOKEN: ${{ secrets.FM_DOCKERHUB_TOKEN }}
+
+      # Pull-image on this stage should only ever have an effect when rerunnning a job from
+      # the github actions web page. This is useful for re-running integration tests to sniff
+      # out intermittent issues.
+      # To rerun the whole build, push a new sha to the branch.
+      - name: Build lite-qserv image
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG build \
+          --pull-image \
+          --push-image
+        env:
+          QSERV_DH_USER: ${{ secrets.FM_DOCKERHUB_TOKEN_USER }}
+          QSERV_DH_TOKEN: ${{ secrets.FM_DOCKERHUB_TOKEN }}
+
+      - name: Launch qserv
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG up \
+
+      - name: Run integration tests
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG itest \
+          --wait 180
+
+      - name: Remove integration test volumes
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG itest-rm \
+
+      - name: Shut down qserv
+        run: |
+          ./admin/local/cli/qserv --log-level DEBUG down \
+          -v \
+
+
+
+
+
+#       - name: Declare Version Variables
```

Comment:
Yes; The GitHub actions CI code was originally written by @fjammes. I changed it, basically entirely, but I didn't want to lose what he had written, at least not right away. As part of our process to keep commits organized & targeted I was afraid it would get lost in a fixup or squash during a rebase.

Probably it should get removed before going to master, or at least an explanation added in code about why the commented code is there. Will do one or the other.